### ISSUE TITLE
Reported MQTT WiFi Signal is not correct

### ISSUE THREAD

**Issue by WolfwithSword** [X1Plus/X1Plus#284]
**Describe the bug**
Since WiFi driver was modified in X1P 1.1, the reported signal strength from the printer's MQTT is hard-stuck at -90dBm. 

As a result, bambustudio/orcaslicer detect low signal strength (does not warn user, but evident in the signal icon) and as a result, will be more likely than not to send print requests via cloud (HTTPS-routing) rather than local (FTP), as can be viewed by monitoring mqtt data for project_file requests.

**To Reproduce**
1. Load X1P 1.1 with new wifi driver changes
2. Monitor MQTT push_status for wifi_signal
3. View it is stuck at -90dBm
4. Optional: Start a print and monitor /request mqtt topic and view it is from cloud.

**Expected behavior**
WiFi signal from MQTT should reflect the proper signal strength. And if it is high enough, printing should prefer to use local FTP instead of cloud service & slicer icon for signal strength will be higher.

**Screenshots/Photos**
![image](https://github.com/X1Plus/X1Plus/assets/12175651/ca20f10d-35fb-457c-aaa5-ef95ba41391e)

![image](https://github.com/X1Plus/X1Plus/assets/12175651/29cefe26-83ac-4c71-8e4f-38d3b2166797)
User in second screenshot had just installed X1P, so their -30dBm value before was pre-X1P 1.1

**External Links**
Has this bug been discussed elsewhere that you can link to? Have you checked this repository's discussions section?

https://discord.com/channels/1041781310299062282/1233015570610524160/1247333181745659996 

**Additional Details:**
 - Printer type (X1/X1C/X1E): X1C
 - X1Plus version: 1.1
 - AP firmware version:
 - AP hardware revision:
 - MC firmware version:
 - MC hardware revision:
 - Toolhead firmware version:
 - Toolhead hardware revision:
 - Computer vision model:

**Is the bug present on OEM firmware?**
Did not test

**Custom modifications**
Additionally monitored via NodeRed to view wifi signal

**Additional context**
It is reported that checking `iw dev wlan0 station dump` reports the correct value.

Copied from @jwise in discord:

```
looks like forward tries to open("/proc/net/wireless")
indeed, in bbl_utils there is a function 493: 000671f8  2108 FUNC    GLOBAL DEFAULT   11 wifi_link_quality[abi:cxx11]()
so probably overriding that is how to do it
```

**Comment by jwise** [comment]
Reported fixed by #285 .

**Comment by WolfwithSword** [comment]
Tested build with that fix - can confirm the signal strength reported in MQTT is now accurate and updates frequently. Also reports as higher in the slicer now.

![image](https://github.com/X1Plus/X1Plus/assets/12175651/c8459817-7a90-4b54-b191-d3c3cb9180cb)
![image](https://github.com/X1Plus/X1Plus/assets/12175651/209c48a7-ec6e-4ae9-a6ff-d1a7d60f2c07)
![image](https://github.com/X1Plus/X1Plus/assets/12175651/f85f1312-3e24-4c5a-8ed0-974dd0854c81)
